koku/api/settings/settings.py

```diff
@@ -228,6 +268,35 @@ def _tag_key_handler(self, settings):
 
         return any(updated)
 
+    def _currency_handler(self, settings):
+        currency = settings
+        try:
+            stored_currency = get_selected_currency_or_setup(self.schema)
+        except Exception as exp:
+            LOG.warning(f"Failed to retrieve currency for schema {self.schema}. Reason: {exp}")
+            return False
+
+        if currency is None or stored_currency == currency:
+            return False
+
+        try:
+            set_currency(self.schema, settings)
+        except Exception as exp:
+            LOG.warning(f"Failed to store new currency settings for schema {self.schema}. Reason: {exp}")
+            return False
```

Comment:
I'm trying to understand why setting the currency would generate an exception, considering it's not a user input?

Reply:
It is not anything the UI will ever hit, it is just good practice when writing any API to make sure you cover yourself with exception handling.

Reply:
Just want to enure it doesn't make it to the browser console